src/main/java/xyz/nucleoid/leukocyte/mixin/rule/FlowableFluidMixin.java

```diff
@@ -0,0 +1,35 @@
+package xyz.nucleoid.leukocyte.mixin.rule;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.fluid.FlowableFluid;
+import net.minecraft.fluid.Fluid;
+import net.minecraft.fluid.FluidState;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.world.BlockView;
+import net.minecraft.world.World;
+import xyz.nucleoid.leukocyte.Leukocyte;
+import xyz.nucleoid.leukocyte.RuleQuery;
+import xyz.nucleoid.leukocyte.rule.ProtectionRule;
+
+@Mixin(FlowableFluid.class)
+public class FlowableFluidMixin {
+    @Inject(method = "canFlow", at = @At("HEAD"), cancellable = true)
```

Comment:
This should probably ideally be injected at the return to avoid running the logic if flow is not possible

Reply:
Fixed in c44877f41c24c6ef2fd57f4ff768109341924921